### PR TITLE
[CodeCamp2023-551] advanced_guides/evaluator.md

### PR DESCRIPTION
**Pull request by YanxingLiu**
add evaluator document_en and fix some errors in evaluator doc_zh_CN

### CODE REVIEW

File: docs/en/advanced_guides/evaluator.md [hunk 1]
```diff
@@ -1,5 +1,220 @@
...
+        if hasattr(metric, 'need_cond_input'):
+            metric_dict['need_cond_input'] = metric.need_cond_input
+        md5 = hashlib.md5(repr(metric_dict).encode('utf-8')).hexdigest()
+        return md5
+```
+
+Finally, this method will generate a sampler for each evaluation metric group and add it to a list to return.
+
+## Evaluation process of an evaluator
+
```
Comment (LeoXing1996) [inline docs/en/advanced_guides/evaluator.md:149]: The implementation of evaluation process can be found in `mmagic.engine.runner.MultiValLoop.run` and `mmagic.engine.runner.MultiTestLoop.run`. Here we take `mmagic.engine.runner.MultiValLoop.run` as example:

**Review by LeoXing1996 — CHANGES_REQUESTED** [review]
Apart from the modification suggestions in the comments, please wrap the code section in doc with ``.

File: docs/en/advanced_guides/evaluator.md (hunk 1 above)
Comment (YanxingLiu) [reply docs/en/advanced_guides/evaluator.md:149]: Thanks for your comments, I have added a paragraph for more clear explaination

**Comment by YanxingLiu** [comment]
> Apart from the modification suggestions in the comments, please wrap the code section in doc with ``.

Thanks, I have carefully checked and wrapped all codes in paragraphs with ``.

File: docs/en/advanced_guides/evaluator.md [hunk 2]
```diff
@@ -1,5 +1,220 @@
...
+- The evaluator is the top-level module for evaluation metrics and usually contains one or more metrics. The purpose of the evaluator is to perform necessary data format conversion and call evaluation metrics to calculate the model accuracy during model evaluation. The evaluator is typically built by an executor or a testing script, which are used for online evaluation and offline evaluation, respectively.
+
+The evaluator in mmagic inherits from the evaluator in mmengine, and the basic usage is similar to the evaluator in mmengine. For specific information, you can refer to [Model Accuracy Evaluation](https://mmengine.readthedocs.io/zh_CN/latest/design/evaluation.html). However, different from other high-level vision tasks, the evaluation metrics for generative models often have multiple inputs. For example, the input for the Inception Score (IS) metric is only fake images and any number of real images, while the Perceptual Path Length (PPL) requires sampling from the latent space. To accommodate different evaluation metrics, mmagic introduces two important methods, "prepare_metrics" and "prepare_samplers," to meet the above requirements.
```
[the remark this answers is not in the document]
Comment (YanxingLiu) [reply docs/en/advanced_guides/evaluator.md]: I have updated the URL,  thanks.

**Review by LeoXing1996 — APPROVED** [review]
LGTM, thanks for your contribution

**Pull request merged**